src/server/classroom/internal.controller.js

```diff
@@ -11,11 +10,14 @@ const APIError = require('../helpers/APIError');
  * @param {import('socket.io').Server} io
  */
 async function sendMessage(packet, socket, io) {
-  console.log('sendMessage receive packet: ');
-  console.log(packet);
+  const [data, callback, meta] = packet;
 
-  const [data, classroomID, meta] = packet;
-  const classroom = Classroom.statics.get(classroomID);
+  let classroom;
+  try {
+    classroom = await Classroom.getCached(data.classroomId);
+  } catch (ex) {
+    return callback({ error: ex.message });
+  }
```

Comment:
The problem is here. Use `meta.invokerClassroom` instead, no need to `getCached` (the middleware already executed `getCached` for you

```suggestion
  if (!meta.invokerClassroom) return callback({ error: 'You are not in a classroom' });
  const classroom = meta.invokerClassroom;
```